Changes were originally part of Solaris bug fix for:
15226832 Fonts (in particular Lucida Sans Typewriter) look terrible in JDS.

Allows fontconfig to use bitmap fonts (which may be hand-tuned for the
specific size) instead of scalable fonts when a close match is found and
anti-aliasing is not requested.

--- fontconfig-2.14.0/src/fcmatch.c
+++ fontconfig-2.14.0/src/fcmatch.c
@@ -50,6 +50,15 @@ default:
 	return -1.0;
     }
     v = v2 - v1;
+
+    /*
+     * Hack to allow bitmap fonts to equal requested font if it is close
+     */
+    if ( v > -.5 && v <= .5 )
+    {
+        v = 0;
+    }
+
     if (v < 0)
 	v = -v;
     *bestValue = FcValueCanonicalize (value2);
@@ -855,6 +864,46 @@ FcConfigSubstituteWithPat (config, new,
     return new;
 }
 
+static void
+FcChangeFormat ( FcPattern *p )
+{
+    int i;
+    FcPatternElt    *pe = FcPatternElts(p);
+
+    /*
+     * Hack. If user want monochrome mode, then try to pick bitmaps over
+     * outlines.
+     */
+    for ( i = 0; i < p->num; i++ )
+    {
+        if ( !strcmp(&pe[i].object, "antialias" ))
+        {
+            int aa = (int) &pe[i].values->value.u.b;
+            int j;
+            int done = 0;
+            FcValue v;
+
+            v.type = FcTypeBool;
+            v.u.b  = aa;
+
+            for ( j = 0; j < p->num; j++ )
+            {
+                if ( !strcmp(&pe[j].object, "outline" ))
+                {
+                    done = 1;
+                    break;
+                }
+            }
+            if ( !done )
+            {
+                FcPatternObjectAddWithBinding (p, "outline", v,
+                        FcValueBindingStrong, 1);
+            }
+            break;
+        }
+    }
+}
+
 static FcPattern *
 FcFontSetMatchInternal (FcFontSet   **sets,
 			int	    nsets,
@@ -870,6 +919,8 @@ int		    set;
     FcCompareData   data;
     const FcPatternElt *elt;
 
+    FcChangeFormat ( p );
+
     for (i = 0; i < PRI_END; i++)
 	bestscore[i] = 0;
     best = 0;
@@ -1035,6 +1086,8 @@ FcResult    *result)
 {
     FcPattern	    *best, *ret = NULL;
 
+    FcChangeFormat ( p );
+
     assert (sets != NULL);
     assert (p != NULL);
     assert (result != NULL);
